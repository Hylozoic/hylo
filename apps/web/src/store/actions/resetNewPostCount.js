import { RESET_NEW_POST_COUNT } from 'store/constants'

export default function resetNewPostCount (id, type, count = 0) {
  if (!['TopicFollow', 'Membership'].includes(type)) {
    throw new Error(`bad type for resetNewPostCount: ${type}`)
  }

  return {
    type: RESET_NEW_POST_COUNT,
    graphql: {
      query: type === 'TopicFollow' ? TopicFollowQuery : MembershipQuery,
      variables: {
        id,
        data: {
          newPostCount: count
        }
      }
    },
    meta: { id, type, count, optimistic: true }
  }
}

const TopicFollowQuery = `mutation($id: ID, $data: TopicFollowInput) {
    updateTopicFollow(id: $id, data: $data) {
      id
      newPostCount
      group {
        id
      }
      topic {
        id
      }
    }
  }`

const MembershipQuery = `mutation($id: ID, $data: MembershipInput) {
    updateMembership(groupId: $id, data: $data) {
      id
    }
  }`
