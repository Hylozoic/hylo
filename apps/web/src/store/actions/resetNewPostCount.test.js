import resetNewPostCount from './resetNewPostCount'

it('throws an error for unknown type', () => {
  let caughtError
  try {
    resetNewPostCount(5, 'GroupTopic')
  } catch (error) {
    caughtError = error.message
  }
  expect(caughtError).toEqual('bad type for resetNewPostCount: GroupTopic')
})

it('works for Membership', () => {
  expect(resetNewPostCount(5, 'Membership')).toMatchSnapshot()
})

it('works for Membership', () => {
  expect(resetNewPostCount(5, 'TopicFollow')).toMatchSnapshot()
})
