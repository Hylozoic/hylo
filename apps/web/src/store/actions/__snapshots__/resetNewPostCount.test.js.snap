// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works for Membership 1`] = `
Object {
  "graphql": Object {
    "query": "mutation($id: ID, $data: MembershipInput) {
    updateMembership(groupId: $id, data: $data) {
      id
    }
  }",
    "variables": Object {
      "data": Object {
        "newPostCount": 0,
      },
      "id": 5,
    },
  },
  "meta": Object {
    "count": 0,
    "id": 5,
    "optimistic": true,
    "type": "Membership",
  },
  "type": "RESET_NEW_POST_COUNT",
}
`;

exports[`works for Membership 2`] = `
Object {
  "graphql": Object {
    "query": "mutation($id: ID, $data: TopicFollowInput) {
    updateTopicFollow(id: $id, data: $data) {
      id
      newPostCount
      group {
        id
      }
      topic {
        id
      }
    }
  }",
    "variables": Object {
      "data": Object {
        "newPostCount": 0,
      },
      "id": 5,
    },
  },
  "meta": Object {
    "count": 0,
    "id": 5,
    "optimistic": true,
    "type": "TopicFollow",
  },
  "type": "RESET_NEW_POST_COUNT",
}
`;
